### PR TITLE
feat: added swagger for api documentation

### LINKED ISSUE #21: Add Swagger for API
Add Swagger to the API to get a overview of all endpoints.

Resource:
https://medium.com/@samuelnoye35/simplifying-api-development-in-node-js-with-swagger-a5021ac45742

### PR DESCRIPTION
**Pull request by OlgaKupar**
The first working version. 

Planned **refactoring**:

 - Rebase Swagger annotations from "user.route.js" and "lobby.route.js" into a single "swagger.js" file.
 - Refactor "user.route.js" and "lobby.route.js" to match the updated Swagger annotations in "swagger.js".

### CODE REVIEW

File: main.js
Comment (OlgaKupar) [inline main.js:1]: fix: to be done in the next commit

File: swagger.js
Comment (OlgaKupar) [inline swagger.js:1]: I would appreciate any ideas on how to auto-format indentations for this specific annotation structure. 

Solution in worst case: remove all indentation spaces (but redability reduction).

Comment (OlgaKupar) [reply swagger.js:1]: solved with npx eslint --fix <path_to_the_file>

File: src/routes/user.route.js [hunk 1]
```diff
@@ -4,9 +4,335 @@ const router = express.Router();
...
+ *     websocket:
+ *      type: string
+ *      description: The connection websocket
+ *    example:
+ *     username: "BestPlayerEver"
+ *     password: "TopSecret&%22?"
+ *     uuid: "9937525"
+ *     registerTimestamp: "2024-04-17T12:30:45.123Z"
+ *     playedGames: ["9994", "2323", "1111"]
+ *     websocket: "ws://example.com/socket"
```
Comment (FRALEXAlexander) [inline src/routes/user.route.js:46]: This should be the id of a socket not a url

File: package.json [hunk 2]
```diff
@@ -32,6 +32,9 @@
     "mongoose": "^8.2.4",
     "node": "^18.20.0",
     "socket.io": "^4.7.5",
+    "swagger": "^0.7.5",
```
Comment (FRALEXAlexander) [inline package.json:35]: This package add 35 vulnerabilities. I do not see where it is used it should be removed

File: src/routes/lobby.route.js [hunk 3]
```diff
@@ -4,9 +4,315 @@ const router = express.Router();
...
+ *               description: The lobby name
+ *             timestamp:
+ *               type: date
+ *               description: The lobby creation timestamp
+ *             players:
+ *               type: [string]
+ *               description: The player´s usernames array
+ *             maxPlayers:
+ *               type: number
+ *               description: The max number of players for the lobby (3, 4 or 6)
```
Comment (FRALEXAlexander) [inline src/routes/lobby.route.js:40]: 5 players is also valid

File: swagger.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const swaggerDefinition = {
+  openapi: '3.0.0',
+  info: {
+    title: 'Wizard API',
+    version: '1.0.0',
+    description: 'Wizard Board Game API',
+  },
+  servers: [
+    {
+      url: 'http://localhost:8080',
```
Comment (FRALEXAlexander) [inline swagger.js:12]: Change this to get the url of the running instance it will not always be localhost:8080

**Review by FRALEXAlexander — CHANGES_REQUESTED** [review]

**Comment by FRALEXAlexander** [comment]
Run your code before pushing to git

**Review by FRALEXAlexander — APPROVED** [review]
LGTM

**Pull request merged**